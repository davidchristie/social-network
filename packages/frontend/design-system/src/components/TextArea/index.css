.TextArea {
  font-size: 1.2em;
  min-height: 50px;
  padding: 5px;
  resize: vertical;
  width: 100%;
}
