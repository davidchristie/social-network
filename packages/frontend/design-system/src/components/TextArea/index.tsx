import TextField from "@material-ui/core/TextField";
import React from "react";

interface Props {
  autoFocus?: boolean;
  className?: string;
  id: string;
  label: string;
  name: string;
  onBlur?: React.FocusEventHandler;
  onChange?: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onClick?: React.MouseEventHandler;
  required?: boolean;
  type?: string;
  value?: number | string;
}

const TextArea: React.StatelessComponent<Props> = ({
  id,
  label,
  name,
  required,
  ...attributes
}) => {
  return (
    <TextField
      autoComplete={name}
      id={id}
      fullWidth={true}
      margin="normal"
      multiline={true}
      name={name}
      {...attributes}
    />
  );
};

export default TextArea;
