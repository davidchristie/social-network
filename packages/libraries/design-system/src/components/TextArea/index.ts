import React from "react";

import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.TextareaHTMLAttributes<HTMLTextAreaElement> { }

const TextArea = elementWithClass<Props>("TextArea", "textarea");

export default TextArea;
