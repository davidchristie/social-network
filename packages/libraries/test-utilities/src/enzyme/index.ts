export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
