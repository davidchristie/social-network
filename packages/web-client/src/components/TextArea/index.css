.TextArea {
  font-size: 1.4em;
  min-height: 100px;
  resize: vertical;
  width: 100%;
}
