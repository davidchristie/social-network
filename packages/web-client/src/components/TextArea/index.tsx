import React from "react";

import "./index.css";

interface Props extends React.TextareaHTMLAttributes<HTMLTextAreaElement> { }

export default class TextArea extends React.Component<Props> {
  public render () {
    const { className, ...attributes } = this.props;
    return (
      <textarea
        className={`TextArea${className ? " " + className : ""}`}
        {...attributes}
      />
    );
  }
}
