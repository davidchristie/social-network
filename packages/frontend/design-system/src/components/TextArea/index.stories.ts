import { action } from "@storybook/addon-actions";
import TextArea from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(TextArea, {
  id: "text-area-id",
  label: "TextArea",
  name: "text-area",
  onBlur: action("onBlur"),
  onChange: action("onChange"),
  onClick: action("onClick"),
});
