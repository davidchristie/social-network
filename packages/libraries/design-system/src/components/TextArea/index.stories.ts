import { action } from "@storybook/addon-actions";

import TextArea from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(TextArea, {
  onBlur: action("onBlur"),
  onChange: action("onChange"),
  onClick: action("onClick"),
});
