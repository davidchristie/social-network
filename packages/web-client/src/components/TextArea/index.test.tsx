import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import TextArea from "./index";

describe("TextArea component", () => {
  let wrapper: ShallowWrapper;

  describe("with no props", () => {
    beforeEach(() => {
      wrapper = shallow(<TextArea />);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("with custom class", () => {
    const CUSTOM_CLASS = "custom_class";

    beforeEach(() => {
      wrapper = shallow(
        <TextArea className={CUSTOM_CLASS} />,
      );
    });

    it("has custom class", () => {
      expect(wrapper.hasClass(CUSTOM_CLASS)).toBe(true);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
