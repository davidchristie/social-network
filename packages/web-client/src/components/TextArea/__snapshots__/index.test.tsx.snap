// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component with custom class matches snapshot 1`] = `
<textarea
  className="TextArea custom_class"
/>
`;

exports[`TextArea component with no props matches snapshot 1`] = `
<textarea
  className="TextArea"
/>
`;
